### PR TITLE
perf tests for create and delete directory

### PR DESCRIPTION
**Pull request by Anipik**
Fixes https://github.com/dotnet/corefx/issues/23759

### CODE REVIEW

File: src/System.IO.FileSystem/tests/Performance/Perf.Directory.cs [hunk 1]
```diff
@@ -55,5 +57,79 @@ public void Exists()
...
+                sb.Append(@"\a");
+            }
+
+            string path = sb.ToString();
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < 1000; i++)
```
Comment (Anipik) [inline src/System.IO.FileSystem/tests/Performance/Perf.Directory.cs]: @JeremyKuhne how to decide the number ```1000```. with 20000 I am getting ```nan``` as output

Comment (JeremyKuhne) [reply src/System.IO.FileSystem/tests/Performance/Perf.Directory.cs]: Whatever runs in a measurable amount of time. You don't need to run forever- just something that gives a large enough time to filter out some of the noise (few hundred ms or larger).  You might want to consider passing in a different count for each of your data points.

**Comment by Anipik** [comment]
System.IO.FileSystem.Performance.Tests.dll                                      | Metric   | Unit | Iterations |   Average | STDEV.S |       Min |       Max
:------------------------------------------------------------------------------- |:-------- |:----:|:----------:| ---------:| -------:| ---------:| ---------:
 System.IO.Tests.Perf_Directory.CreateDirectory                                  | Duration | msec |     1      | 10895.520 |     NaN | 10895.520 | 10895.520
 System.IO.Tests.Perf_Directory.RecursiveCreateDirectory(depth: 10, times: 2000) | Duration | msec |     1      |  7359.714 |     NaN |  7359.714 |  7359.714
 System.IO.Tests.Perf_Directory.RecursiveCreateDirectory(depth: 100, times: 200) | Duration | msec |     1      |  8874.217 |     NaN |  8874.217 |  8874.217
 System.IO.Tests.Perf_Directory.RecursiveCreateDirectory(depth: 1000, times: 20) | Duration | msec |     1      | 28016.476 |     NaN | 28016.476 | 28016.476
 System.IO.Tests.Perf_Directory.RecursiveDeleteDirectory(depth: 10, times: 2000) | Duration | msec |     1      |  6800.055 |     NaN |  6800.055 |  6800.055
 System.IO.Tests.Perf_Directory.RecursiveDeleteDirectory(depth: 100, times: 200) | Duration | msec |     1      |  8438.337 |     NaN |  8438.337 |  8438.337
 System.IO.Tests.Perf_Directory.RecursiveDeleteDirectory(depth: 1000, times: 20) | Duration | msec |     1      | 30372.156 |     NaN | 30372.156 | 30372.156

**Comment by danmosemsft** [comment]
Have you verified that this gives durations that are within the recommended range (100ms-1sec I belive)

**Comment by Anipik** [comment]
System.IO.FileSystem.Performance.Tests.dll                                                        | Metric   | Unit | Iterations |   Average | STDEV.S |       Min |       Max
:------------------------------------------------------------------------------------------------- |:-------- |:----:|:----------:| ---------:| -------:| ---------:| ---------:
 System.IO.Tests.Perf_Directory.CreateDirectory                                                    | Duration | msec |     1      | 12487.011 |     NaN | 12487.011 | 12487.011
 System.IO.Tests.Perf_Directory.Exists                                                             | Duration | msec |     3      |  5141.609 | 323.231 |  4857.087 |  5493.068
 System.IO.Tests.Perf_Directory.GetCurrentDirectory                                                | Duration | msec |    383     |    26.062 |  10.725 |    21.840 |   203.814
 System.IO.Tests.Perf_Directory.RecursiveCreateDirectory(depth: 10, times: 1000)                   | Duration | msec |     2      |  4080.509 | 103.606 |  4007.248 |  4153.769
 System.IO.Tests.Perf_Directory.RecursiveCreateDirectory(depth: 100, times: 100)                   | Duration | msec |     2      |  4711.934 |   2.211 |  4710.370 |  4713.497
 System.IO.Tests.Perf_Directory.RecursiveCreateDirectory(depth: 1000, times: 10)                   | Duration | msec |     1      | 15872.034 |     NaN | 15872.034 | 15872.034
 System.IO.Tests.Perf_Directory.RecursiveDeleteDirectory(depth: 10, times: 1000)                   | Duration | msec |     2      |  3931.073 |  13.962 |  3921.201 |  3940.946
 System.IO.Tests.Perf_Directory.RecursiveDeleteDirectory(depth: 100, times: 100)                   | Duration | msec |     2      |  4101.075 |  30.742 |  4079.337 |  4122.813
 System.IO.Tests.Perf_Directory.RecursiveDeleteDirectory(depth: 1000, times: 10)                   | Duration | msec |     1      | 15087.555 |     NaN | 15087.555 | 15087.555
 System.IO.Tests.Perf_File.Delete                                                                  | Duration | msec |     3      |  1226.239 |  43.849 |  1191.800 |  1275.601
 System.IO.Tests.Perf_File.Exists                                                                  | Duration | msec |     2      |  5190.539 |  32.460 |  5167.587 |  5213.492
 System.IO.Tests.Perf_FileInfo.ctor_str                                                            | Duration | msec |    115     |    87.156 |   7.250 |    80.634 |   148.087
 System.IO.Tests.Perf_FileStream.Read(useAsync: False, bufferSize: 1, totalSize: 10000000)         | Duration | msec |    110     |    91.582 |   5.360 |    71.340 |   102.858
 System.IO.Tests.Perf_FileStream.Read(useAsync: False, bufferSize: 1, totalSize: 200000)           | Duration | msec |    1000    |     1.901 |   0.186 |     1.031 |     2.413
 System.IO.Tests.Perf_FileStream.Read(useAsync: False, bufferSize: 10000000, totalSize: 10000000)  | Duration | msec |    1000    |     3.046 |   0.365 |     2.704 |     7.380
 System.IO.Tests.Perf_FileStream.Read(useAsync: False, bufferSize: 200000, totalSize: 200000)      | Duration | msec |    1000    |     0.029 |   0.003 |     0.021 |     0.042
 System.IO.Tests.Perf_FileStream.Read(useAsync: False, bufferSize: 512, totalSize: 10000000)       | Duration | msec |    875     |    11.248 |   2.215 |     7.752 |    27.364
 System.IO.Tests.Perf_FileStream.Read(useAsync: False, bufferSize: 512, totalSize: 200000)         | Duration | msec |    1000    |     0.190 |   0.012 |     0.177 |     0.359
 System.IO.Tests.Perf_FileStream.Read(useAsync: True, bufferSize: 10000000, totalSize: 10000000)   | Duration | msec |    1000    |     3.652 |   0.098 |     3.181 |     4.345
 System.IO.Tests.Perf_FileStream.Read(useAsync: True, bufferSize: 200000, totalSize: 200000)       | Duration | msec |    1000    |     0.054 |   0.009 |     0.043 |     0.180
 System.IO.Tests.Perf_FileStream.Read(useAsync: True, bufferSize: 512, totalSize: 10000000)        | Duration | msec |    161     |    62.061 |   8.438 |    50.454 |   125.192
 System.IO.Tests.Perf_FileStream.Read(useAsync: True, bufferSize: 512, totalSize: 200000)          | Duration | msec |    1000    |     1.414 |   0.444 |     1.150 |     6.032
 System.IO.Tests.Perf_FileStream.Write(useAsync: False, bufferSize: 1, totalSize: 10000000)        | Duration | msec |    118     |    79.775 |  17.612 |    64.812 |   248.331
 System.IO.Tests.Perf_FileStream.Write(useAsync: False, bufferSize: 1, totalSize: 200000)          | Duration | msec |    944     |     1.899 |   0.414 |     1.298 |     3.047
 System.IO.Tests.Perf_FileStream.Write(useAsync: False, bufferSize: 10000000, totalSize: 10000000) | Duration | msec |    998     |     3.917 |   0.348 |     3.200 |     5.233
 System.IO.Tests.Perf_FileStream.Write(useAsync: False, bufferSize: 200000, totalSize: 200000)     | Duration | msec |    1000    |     0.175 |   0.078 |     0.122 |     2.138
 System.IO.Tests.Perf_FileStream.Write(useAsync: False, bufferSize: 512, totalSize: 10000000)      | Duration | msec |    366     |    21.915 |   5.504 |    15.054 |    64.316
 System.IO.Tests.Perf_FileStream.Write(useAsync: False, bufferSize: 512, totalSize: 200000)        | Duration | msec |    1000    |     0.693 |   0.233 |     0.408 |     6.440
 System.IO.Tests.Perf_FileStream.Write(useAsync: True, bufferSize: 10000000, totalSize: 10000000)  | Duration | msec |    895     |     4.254 |   0.360 |     3.522 |    10.831
 System.IO.Tests.Perf_FileStream.Write(useAsync: True, bufferSize: 200000, totalSize: 200000)      | Duration | msec |    1000    |     0.357 |   0.085 |     0.230 |     1.431
 System.IO.Tests.Perf_FileStream.Write(useAsync: True, bufferSize: 512, totalSize: 10000000)       | Duration | msec |     86     |   109.244 |   7.015 |    96.830 |   128.354
 System.IO.Tests.Perf_FileStream.Write(useAsync: True, bufferSize: 512, totalSize: 200000)         | Duration | msec |    1000    |     2.342 |   0.448 |     1.728 |    11.725

**Comment by Anipik** [comment]
@danmosemsft they are arounf 4secs should I reduce it further ?

**Comment by danmosemsft** [comment]
Probably...see the perf testing documentation in the repo.

We know that some preexisting tests already have poorly chosen iteration durations but new ones shouldn't

**Comment by Anipik** [comment]
System.IO.FileSystem.Performance.Tests.dll                                                        | Metric   | Unit | Iterations |   Average | STDEV.S |       Min |       Max
:------------------------------------------------------------------------------------------------- |:-------- |:----:|:----------:| ---------:| -------:| ---------:| ---------:
 System.IO.Tests.Perf_Directory.CreateDirectory                                                    | Duration | msec |     1      | 11100.952 |     NaN | 11100.952 | 11100.952
 System.IO.Tests.Perf_Directory.Exists                                                             | Duration | msec |     2      |  5133.428 | 290.914 |  4927.721 |  5339.135
 System.IO.Tests.Perf_Directory.GetCurrentDirectory                                                | Duration | msec |    384     |    26.057 |   5.541 |    23.399 |    67.491
 System.IO.Tests.Perf_Directory.RecursiveCreateDirectory(depth: 10, times: 100)                    | Duration | msec |     14     |   383.849 |  21.016 |   362.561 |   436.999
 System.IO.Tests.Perf_Directory.RecursiveCreateDirectory(depth: 100, times: 10)                    | Duration | msec |     10     |   536.583 |  56.059 |   481.580 |   627.340
 System.IO.Tests.Perf_Directory.RecursiveCreateDirectory(depth: 1000, times: 1)                    | Duration | msec |     5      |  1311.332 | 422.458 |   773.567 |  1674.138
 System.IO.Tests.Perf_Directory.RecursiveDeleteDirectory(depth: 10, times: 100)                    | Duration | msec |     14     |   379.289 |  20.572 |   349.763 |   428.122
 System.IO.Tests.Perf_Directory.RecursiveDeleteDirectory(depth: 100, times: 10)                    | Duration | msec |     12     |   416.270 |  24.465 |   386.019 |   462.628
 System.IO.Tests.Perf_Directory.RecursiveDeleteDirectory(depth: 1000, times: 1)                    | Duration | msec |     5      |   878.455 | 348.225 |   440.360 |  1275.334
 System.IO.Tests.Perf_File.Delete                                                                  | Duration | msec |     3      |  1149.367 |   9.240 |  1142.046 |  1159.749
 System.IO.Tests.Perf_File.Exists                                                                  | Duration | msec |     2      |  5103.701 | 211.030 |  4954.480 |  5252.921
 System.IO.Tests.Perf_FileInfo.ctor_str                                                            | Duration | msec |    118     |    85.157 |   7.934 |    79.773 |   133.566
 System.IO.Tests.Perf_FileStream.Read(useAsync: False, bufferSize: 1, totalSize: 10000000)         | Duration | msec |    166     |    60.376 |   4.038 |    57.410 |    82.033
 System.IO.Tests.Perf_FileStream.Read(useAsync: False, bufferSize: 1, totalSize: 200000)           | Duration | msec |    1000    |     1.377 |   0.971 |     1.104 |    21.481
 System.IO.Tests.Perf_FileStream.Read(useAsync: False, bufferSize: 10000000, totalSize: 10000000)  | Duration | msec |    1000    |     2.936 |   0.146 |     2.784 |     3.903
 System.IO.Tests.Perf_FileStream.Read(useAsync: False, bufferSize: 200000, totalSize: 200000)      | Duration | msec |    1000    |     0.023 |   0.003 |     0.021 |     0.037
 System.IO.Tests.Perf_FileStream.Read(useAsync: False, bufferSize: 512, totalSize: 10000000)       | Duration | msec |    1000    |     8.136 |   0.827 |     7.535 |    22.007
 System.IO.Tests.Perf_FileStream.Read(useAsync: False, bufferSize: 512, totalSize: 200000)         | Duration | msec |    1000    |     0.127 |   0.016 |     0.117 |     0.252
 System.IO.Tests.Perf_FileStream.Read(useAsync: True, bufferSize: 10000000, totalSize: 10000000)   | Duration | msec |    1000    |     2.929 |   0.160 |     2.716 |     4.431
 System.IO.Tests.Perf_FileStream.Read(useAsync: True, bufferSize: 200000, totalSize: 200000)       | Duration | msec |    1000    |     0.049 |   0.007 |     0.042 |     0.106
 System.IO.Tests.Perf_FileStream.Read(useAsync: True, bufferSize: 512, totalSize: 10000000)        | Duration | msec |    191     |    52.311 |   8.935 |    47.366 |   152.175
 System.IO.Tests.Perf_FileStream.Read(useAsync: True, bufferSize: 512, totalSize: 200000)          | Duration | msec |    1000    |     0.990 |   0.103 |     0.863 |     2.042
 System.IO.Tests.Perf_FileStream.Write(useAsync: False, bufferSize: 1, totalSize: 10000000)        | Duration | msec |    142     |    65.618 |  10.956 |    59.833 |   173.170
 System.IO.Tests.Perf_FileStream.Write(useAsync: False, bufferSize: 1, totalSize: 200000)          | Duration | msec |    981     |     1.815 |   0.388 |     1.243 |     2.938
 System.IO.Tests.Perf_FileStream.Write(useAsync: False, bufferSize: 10000000, totalSize: 10000000) | Duration | msec |    1000    |     3.333 |   0.375 |     3.086 |    13.474
 System.IO.Tests.Perf_FileStream.Write(useAsync: False, bufferSize: 200000, totalSize: 200000)     | Duration | msec |    1000    |     0.138 |   0.023 |     0.105 |     0.481
 System.IO.Tests.Perf_FileStream.Write(useAsync: False, bufferSize: 512, totalSize: 10000000)      | Duration | msec |    471     |    16.467 |   2.520 |    13.130 |    33.442
 System.IO.Tests.Perf_FileStream.Write(useAsync: False, bufferSize: 512, totalSize: 200000)        | Duration | msec |    1000    |     0.544 |   0.989 |     0.337 |    31.246
 System.IO.Tests.Perf_FileStream.Write(useAsync: True, bufferSize: 10000000, totalSize: 10000000)  | Duration | msec |    1000    |     3.468 |   0.230 |     3.198 |     5.209
 System.IO.Tests.Perf_FileStream.Write(useAsync: True, bufferSize: 200000, totalSize: 200000)      | Duration | msec |    1000    |     0.251 |   0.045 |     0.193 |     0.555
 System.IO.Tests.Perf_FileStream.Write(useAsync: True, bufferSize: 512, totalSize: 10000000)       | Duration | msec |    112     |    84.822 |   3.467 |    78.513 |    97.010
 System.IO.Tests.Perf_FileStream.Write(useAsync: True, bufferSize: 512, totalSize: 200000)         | Duration | msec |    1000    |     2.087 |   0.265 |     1.549 |     5.061

**Comment by Anipik** [comment]
@dotnet-bot test Windows x86 Release Build

**Pull request merged**